### ISSUE TITLE
Contract call inside another contract failed

### ISSUE THREAD

**Issue by NGDFu** [ethereum/go-ethereum#19104]
Hi there,

2 contracts, Called and Test, And 1 private chain with geth's version 1.8.20. I only deployed the Test, and Called is newed by Test(See code for detail).
The question is, when i call setHello("Hello World") at Test, it's all OK, but after that, when i call getHello(), get empty string. But in Called, getHello() returns "Hello World".

All steps:
    In Test:
    1. setHello("Hello World") -> Success
    **2. getHello() -> Sucess, but returns empty string** (Why ???)
    3. getCallAddress() -> Success, get the Called's address
    So, In Called:
    4. getHello() -> Success, and returns "Hello World"

**I am use [Remix](http://remix.ethereum.org) for test.
Use Ropsten is OK. Only with problem in private chain.**

```
pragma solidity 0.5.3;

interface ICall {
    function setHello(string calldata _hello) external;
    function getHello() external view returns (string memory);
    function setI(uint256 _i) external;
    function getI() external view returns (uint256);
}

contract Called is ICall {
    string public hello;

    uint256 public i;

    function setHello(string calldata _hello) external {
        hello = _hello;
    }

    function getHello() public view returns (string memory) {
        return hello;
    }

    function setI(uint256 _i) external {
        i = _i;
    }

    function getI() external view returns (uint256) {
        return i;
    }
}

contract Test {
    string strHello;

    ICall public called = new Called();

    uint256 constant BASE_MASK = uint256(uint128(~0)) << 128;       // highest 128 bits for Base Type;
    uint256 constant NFT_INDEX_MASK = uint256(uint64(~0)) << 64;
    uint256 constant FT_INDEX_MASK = uint256(uint64(~0));

    uint256 public iTest = 30;

    event LogSetHello(address indexed owner, string hello);
    event LogSetI(address indexed onwer, uint256 i);
    
    function getCallAddress() public view returns (address) {
        return address(called);
    }

    function getHello() public view returns (string memory _helloTmp) {
        _helloTmp = called.getHello();
    }

    function setHello(string calldata _hello) external {
        called.setHello(_hello);

        emit LogSetHello(msg.sender, _hello);
    }
}
```

**Comment by holiman** [comment]
How do you call setHello? Using 'eth_call' or by actually sending a transaction that gets mined on chain? The former just evaluates the call on the local node, doesn't affect state

**Comment by NGDFu** [comment]
@holiman I am use Remix, so it's just sending a transaction.

**Comment by marcosmartinez7** [comment]
Did you check out that the remix transacction succeeded?

**Comment by holiman** [comment]
Works perfectly for me, the exact code above.. 
![Screenshot_2019-06-19 Remix - Ethereum IDE](https://user-images.githubusercontent.com/142290/59760800-71497c80-9293-11e9-9281-08f4b3835e9c.png)

**Comment by holiman** [comment]
Please reopen if still relevant -- but if so, please also add more info about how you invoke it. Otherwise it's hard to follow along and understand what is wrong.
